FEAT: Kimi-VL-A3B

Minamiyama opened the pull request.

Minamiyama committed:
feat: add Kimi-VL model cards

Minamiyama committed:
feat(vllm): 添加对Kimi-VL-A3B-Instruct和Kimi-VL-A3B-Thinking视觉模型的支持

Minamiyama committed:
feat: [WIP]transformers impl

Minamiyama committed:
[WIP]model vision ability

Minamiyama committed:
[WIP]generate

MilongWong commented:
Just found this info after I failed to try the Kimi-VL by Xinference. Is there any plan to make it merged?

MilongWong commented:
I just tried in the xinference latest docker with cmd below, and seems that it works  (on 1 H20 96G): 

python3 -m vllm.entrypoints.openai.api_server --port 8888 --served-model-name kimi-vl --trust-remote-code --model /workspace/modelscope/hub/models/moonshotai/Kimi-VL-A3B-Thinking/ --tensor-parallel-size 1 --max-num-batched-tokens 131072 --max-model-len 131072 --max-num-seqs 512 --limit-mm-per-prompt image=256

Minamiyama commented:
> Just found this info after I failed to try the Kimi-VL by Xinference. Is there any plan to make it merged?

80% WIP, waiting for a refactor to multimodal LLM engine

Minamiyama committed:
Merge branch 'main' into FEAT/model/llm/Kimi-VL-A3B

# Conflicts:
#	xinference/model/llm/llm_family.json
#	xinference/model/llm/llm_family_modelscope.json

Minamiyama committed:
feat(model): add Kimi-VL-A3B models to llm_family.json

Add support for Kimi-VL-A3B-Instruct and Kimi-VL-A3B-Thinking-2506 vision-language models with multimodal reasoning capabilities

Minamiyama committed:
fix(vllm): update Kimi-VL-A3B-Thinking model name to include version
